flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OutputWithRecordsCountCheck.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.tasks;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.util.OutputTag;
+
+/** This is a wrapper for outputs to check whether its record output needs to be counted. */
+@Internal
+public interface OutputWithRecordsCountCheck<OUT> {
```

Comment:
Good point! But I made it inherit `WatermarkGaugeExposingOutput` instead of `Output` as the return value of `org.apache.flink.streaming.runtime.tasks.OperatorChain#createOutputCollector` is `WatermarkGaugeExposingOutput`, which is easier to handle. WDTY？